vsts/pipelines/_buildimage-bases.job-template.yml

```diff
@@ -0,0 +1,39 @@
+parameters:
+  name: ''
+  displayName: ''
+  imageDir: ''
+
+
+jobs:
+- job: ${{ parameters.name }}
+  displayName: ${{ parameters.displayName }}
+  steps:
+
+  - task: ShellScript@2
+    displayName: Build images
+    inputs:
+      scriptPath: ./build/build-buildimage-bases.sh
+      args: ${{ parameters.imageDir }}
+
+  - task: CopyFiles@2
+    displayName: Copy artifacts to staging directory
+    inputs:
+      sourceFolder: '$(Build.SourcesDirectory)/artifacts'
+      contents: '**/*.*'
+      targetFolder: $(Build.ArtifactStagingDirectory)
+      overWrite: true
+    condition: true
+
+  - task: Docker@1
+    displayName: Push built images to dev ACR
+    inputs:
+      command: push
+      azureSubscriptionEndpoint: $(ascName)
+      azureContainerRegistry: $(acrName)
+      pushMultipleImages: true
+      imageNamesPath: '$(Build.ArtifactStagingDirectory)/images/build-image-bases.txt'
```

Comment:
There could be a potential race condition here since the same file name is used by parallel build jobs (for example, a python build could overwrite this file which had content written by a php build). We should rather have unique file names which are based on the 'image directory' being built.

Reply:
Thinking more, looks like we are not going to hit the race condition issue here as `$(Build.ArtifactStagingDirectory)` resolves to a directory on the build agent itself and not on some common share. But there would race condition for the below task `Publish build artifacts` where parallel jobs would override the file with the same name.